tests/connection_utils.py

```diff
@@ -29,9 +29,12 @@ async def disconnect_all(server: ChiaServer) -> None:
     for con in cons:
         await con.close()
 
+    await asyncio.sleep(5)  # 5 seconds to allow connections and tasks to all drain
+
```

Comment:
Can we monitor for the desired state instead of just sleeping and hoping?

Reply:
I'd like to but struggled to find a way.

Reply:
It seems that `WSChiaConnection.close()` is not great.  It cancels tasks without actually waiting for them to finish.  Add another PR to the list...  if you want to maybe save some time you could gather all the `.close()` calls concurrently.  Also not sure why we're slicing the fresh list either.

```python
    async def disconnect_all(server: ChiaServer) -> None:
        connections = list(server.all_connections.values())
        await asyncio.gather(*(connection.close() for connection in connections))
        await asyncio.sleep(5)  # 5 seconds to allow connections and tasks to all drain
```

I also made a note over in https://github.com/Chia-Network/chia-blockchain/pull/13778 to make this timeout adjusted.

Reply:
I think it's a little more complicated since WSChiaConnection doesn't know about all the tasks in use - I don't think we should be calling the close method directly on the connections, this should be done at the service/client level since there we can wait on all the tasks.

Reply:
Yeah, there probably is more.  That's what happens with _un_structured concurrency.  You can't really tell if something is done or not.